sdk/lib/_internal/js_runtime/lib/js_helper.dart

```diff
@@ -378,6 +378,14 @@ class Primitives {
     return result;
   }
 
+  static bool? parseBool(String source) {    
+    if (source == null) throw argumentErrorValue(source);
+    if (source.isEmpty) return null;
+    source = source.toLowerCase();
+    if (source != 'true' || source != 'false') return null;                      
+    return (str == 'true') ? true : false;
```

Comment:
Extra comparison to `true` isn't necessart.

For the case-sensitive version:
```dart
static bool? parseBool(String source) => 
    source == "true" ? true : source == "false" ? false : null;
```
(The `==` is again expected to be efficient and check `length` first.)